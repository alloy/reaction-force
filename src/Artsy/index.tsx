export * from "./Analytics"
export { ContextProvider, ContextConsumer, ContextProps } from "./SystemContext"
