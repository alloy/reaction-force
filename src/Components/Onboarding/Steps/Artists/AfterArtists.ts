// This is a list of the first 100 after artists sorted by follow_count.
// The looker SQL that generates this list could be found here:
//   https://artsy.looker.com/sql/djr8sptb6d7dvt
export const AfterArtists = [
  { id: "561028b77261697ddc00021b", name: "After Andy Warhol" },
  { id: "57d9a1c59c18db6b61000b05", name: "After Jean-Michel Basquiat" },
  { id: "58a37e53a09a6771d84d9ba1", name: "After Roy Lichtenstein" },
  { id: "56e078c99c18db3e520023f6", name: "After Alexander Calder" },
  { id: "515b34ab223afa29bd000082", name: "after Leonardo da Vinci" },
  { id: "58a37d959c18db1fb041817e", name: "After Marc Chagall" },
  { id: "56d895ffcd530e657c0008db", name: "After Katsushika Hokusai" },
  { id: "50ed855267f4ee56af000a73", name: "After Jean-Marc Nattier" },
  { id: "5928b21fa09a675e1eb3a4bf", name: "After Jackson Pollock" },
  { id: "515b44fc9562c8fee20022ce", name: "After Auguste Rodin" },
  { id: "59e6328b2a893a0c4fcfc7f7", name: "After L S Lowry" },
  { id: "515d62d45eeb1c904c0046e9", name: "After Michelangelo" },
  { id: "515b34a8223afa29bd000066", name: "After Pieter Bruegel the Elder" },
  { id: "515cdd197b70570a13000ecf", name: "After Albrecht Dürer" },
  { id: "515b045e05635113a5000270", name: "after Henri de Toulouse-Lautrec" },
  { id: "515d62d45eeb1c904c0046f0", name: "after Hans Vredeman de Vries" },
  { id: "515b1aca38ad2d78ca000b99", name: "After Jean-Honoré Fragonard" },
  { id: "515b043305635113a500024d", name: "after Jacques Callot" },
  { id: "5abcda96139b2124d2e949ca", name: "After KAWS" },
  { id: "50ed84b967f4ee56af000a5d", name: "After Bernardino Luini" },
  { id: "515ba3c8cd4b8ed0b9000d5d", name: "After Parmigianino" },
  { id: "515b2ae7223afa9b3e00014f", name: "after Lucas Cranach the Younger" },
  { id: "59fa25eca09a6703be8e739a", name: "After Austin Osman Spare" },
  { id: "515bafe4ea092f003c000e9c", name: "after Correggio" },
  { id: "59ea0c698b3b8126cefed1e6", name: "After Andy Warhol X The Skateroom" },
  { id: "50ed8546468c145c060009f9", name: "After Jan Sadeler I" },
  { id: "50ed855ed006a38e4a000a7a", name: "After Rembrandt van Rijn" },
  { id: "515b279c38ad2d3bc50001d2", name: "after Antoine Watteau" },
  {
    id: "515b179538ad2dfc77000a99",
    name: "after Elisabeth-Louise Vigée Le Brun",
  },
  { id: "515b318e05635113a5001b44", name: "after Charles Le Brun" },
  { id: "515ba015ea092ff5790005d3", name: "after Claude Lorrain" },
  { id: "515bb3231b12b0244a001da3", name: "After Giambologna" },
  { id: "515bc2891b12b0e6680026b7", name: "after Andrea del Sarto" },
  { id: "515cdd16769659e1fb000eb2", name: "after Donatello" },
  { id: "515d0ea65eeb1c904c002632", name: "after Hans Baldung Grien" },
  { id: "515b144138ad2d78ca0007e1", name: "after Baccio Baldini" },
  { id: "515b112b223afae9a500061d", name: "After Léon Cogniet" },
  { id: "515b1ac99562c8fee2000c87", name: "after Thomas Rowlandson" },
  { id: "515b34a99562c8fee2001ab5", name: "after Hans Sebald Beham" },
  { id: "515baa0a94714c2e38000f08", name: "after Polidoro da Caravaggio" },
  { id: "515cccbb5eeb1c524c0005ee", name: "after Lodovico Carracci" },
  { id: "515bb61fea092f003c001183", name: "After Andrea Mantegna" },
  { id: "5ac19e74139b2124d2ebe8ab", name: "After Christopher Wool" },
  { id: "515d3a01b5907b33b10038a0", name: "after Andrea del Verrocchio" },
  { id: "515b0433223afa7ea8000056", name: "after Maerten van Heemskerck" },
  { id: "55523b8c726169115d640000", name: "After a design by Fontana Arte" },
  { id: "515b34ac38ad2dfb2f0005ea", name: "After Maarten de Vos" },
  {
    id: "59af152c8b0c147d8296d661",
    name: "After John Dixon \u0026 George Stubbs by Robert Laurie",
  },
  { id: "515b450f05635113a5002490", name: "after Pietro da Cortona" },
  {
    id: "53ee760f63617219ae380100",
    name: "After Pierre Joseph Redouté and de Gouy",
  },
  { id: "515b14469562c86fb8000937", name: "after Winslow Homer" },
  { id: "5abea971139b21255e3f16f7", name: "after Pablo Picasso" },
  {
    id: "515d02b55eeb1c904c001fd3",
    name: "after Giovanni Francesco Barbieri, called Guercino",
  },
  {
    id: "515b27999562c86fb80013f4",
    name: "after Léon Cogniet after Théodore Gericault",
  },
  { id: "516cc260b31e2b8bf6000275", name: "after Jacopo Tintoretto" },
  { id: "515b5afc9562c8fee2002b88", name: "after Bertoia" },
  { id: "5a20e634a09a675e0609ce73", name: "After Alexander Wilson" },
  { id: "515ba3ca1b12b0e6680015d5", name: "After Paolo Veronese" },
  { id: "5a0106622a893a6c319d447f", name: "After Titian" },
  { id: "515b0aff223afa7ea8000326", name: "after Bartolomeo Spani" },
  { id: "515b279905635113a500163f", name: "after Daniele Ricciarelli" },
  { id: "515b37ed223afa29bd0001ff", name: "After Raphael" },
  { id: "515b34ac38ad2dfb2f0005f1", name: "after Orazio Gentileschi" },
  { id: "515b4e59223afa29bd000dcb", name: "after Herman van Swanevelt" },
  { id: "515b450838ad2dfb2f000dfe", name: "after Francesco Salviati" },
  { id: "515b665f94714cb2e30000d3", name: "after Wenzel von Olmutz" },
  { id: "515b805bcd4b8ed0b9000387", name: "after Jane Collyer" },
  { id: "515b96f1ea092ff5790002cb", name: "after Jean-Baptiste Joseph Pater" },
  {
    id: "515cd362b5907bf7e800094a",
    name: "after Louis-Pierre Froment after Honoré Daumier",
  },
  { id: "515cd3617b7057eb4c000a0e", name: "after Master IB" },
  { id: "515cc5b47b7057eb4c0003b3", name: "after Israhel van Meckenem" },
  { id: "515cdd185eeb1c904c000d4a", name: "after Jacques Bellange" },
  { id: "515cefb27696593fde0017e3", name: "after Charles-Nicolas Cochin II" },
  { id: "515d0bb0b5907bf7e8002436", name: "after Maso Finiguerra" },
  { id: "515d2ed75eeb1c524c0035bc", name: "after Henri Nicolas van Gorp" },
  {
    id: "515d28927696593fde0033eb",
    name: "after Jean-Jacques-François Le Barbier I",
  },
  { id: "515d5a315eeb1c904c0043bf", name: "after Pietro Testa" },
  { id: "515d2b897b70570a130035df", name: "after Master FVB" },
  { id: "516def4eed1712d0a80006d2", name: "after Johann Gottfried Schadow" },
  { id: "516de80ded1712d0a8000409", name: "after Marco Ricci" },
  { id: "59c14e1f8b0c14109f7db57c", name: "After Christoph Murer" },
  { id: "59bc4140a09a672fdae2441b", name: "After Polykleitos" },
  { id: "59c15af38b0c14109f7db65f", name: "After Robert Adam" },
  { id: "59c15b9eb202a34d77c23298", name: "After Januarius Zick" },
  { id: "59d6d8777622dd73dec418ad", name: "After Claude Vignon" },
  { id: "59c16a77b202a34d77c23431", name: "After Jost Amman" },
  { id: "59d6d24fb202a3703b1f391f", name: "After Antoine Lebel" },
  { id: "59c276f0a09a677e60e9e14e", name: "After John Constable" },
  { id: "59d78d04275b2478510714dd", name: "After Hans Bol" },
  { id: "59d78e4a8b3b81432ad1eda8", name: "After Bartholomaeus Spranger" },
  { id: "59d7947b2a893a01eb2b68ef", name: "After Paul Bril" },
  { id: "59d7902b9c18db249fcc6720", name: "After Joseph Heintz the Elder" },
  { id: "59d78f787622dd4d8710a7d1", name: "After Federico Barocci" },
  { id: "5a0107d8275b24501056e932", name: "After Luca Penni" },
  { id: "5a00f73e2a893a6bae87b273", name: "After Roelandt Savery" },
  {
    id: "5a385d81b202a351892ace0b",
    name: "After Raffaellino da Reggio (Raffaello Motta)",
  },
  { id: "5a20eb5fc9dc240678d1bfa4", name: "After Sébastien Bourdon" },
  { id: "5a3861cfa09a6776720d9305", name: "After Francis Wheatley" },
  { id: "5a20e4639c18db68c961e6ef", name: "After Peter Lely" },
  { id: "5a38687db202a3504fed30f6", name: "After François Girardon" },
]
