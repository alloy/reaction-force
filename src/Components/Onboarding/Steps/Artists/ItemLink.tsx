import * as React from "react"
import { createFragmentContainer, graphql } from "react-relay/compat"
import styled, { StyledFunction } from "styled-components"

import { fadeIn, fadeOut } from "../../../../Assets/Animations"
import * as fonts from "../../../../Assets/Fonts"

import Icon from "../../../Icon"

import UpdateArtistFollowMutation from "./Mutations/UpdateArtistFollow"

const anchor: StyledFunction<State & React.HTMLProps<HTMLInputElement>> = styled.a
const Link = anchor`
  display: flex;
  font-size: 14px;
  color: black;
  text-decoration: none;
  text-transform: uppercase;
  font-family: ${fonts.primary.fontFamily};
  border-top: 1px solid #e5e5e5;
  &:hover {
    background-color: #f8f8f8;
  }
  ${props => (props.fadeIn ? fadeIn : null)}
  ${props => (props.fadeOut ? fadeOut : null)}
`

const Avatar = styled.img`
  padding: 10px 15px 10px 10px;
`

const FullWidthCol = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
`

const Col = styled.div`
  display: flex;
  align-items: center;
`

export interface State {
  fadeIn: boolean
  fadeOut: boolean
  artist_id: string
}

type Props = React.HTMLProps<HTMLAnchorElement> & RelayProps

class ItemLink extends React.Component<Props, State> {
  state = {
    fadeIn: false,
    fadeOut: false,
    artist_id: this.props.artist.id,
  }

  followArtist() {
    // const mutation = new UpdateArtistFollowMutation(this.state)

    // Relay.Store.commitUpdate(mutation, {
    //   onFailure: this.followArtistFailed,
    //   onSuccess: response => {
    //     // do something
    //   },
    // })
    console.log("follow artist!")
  }

  followArtistFailed() {
    null
  }

  onClick() {
    this.followArtist()
  }

  render() {
    return (
      <Link
        onClick={this.onClick.bind(this)}
        fadeIn={this.state.fadeIn}
        fadeOut={this.state.fadeOut}
        artist_id={this.props.artist.id}
      >
        <Col>
          {<Avatar src={this.props.artist.image ? this.props.artist.image.cropped.url : ""} width={50} height={50} />}
        </Col>
        <FullWidthCol>{this.props.artist.name}</FullWidthCol>
        <Col>
          <Icon name="follow-circle" color="black" fontSize="39px" />
        </Col>
      </Link>
    )
  }
}

export default createFragmentContainer(
  ItemLink,
  graphql`
    fragment ItemLink_artist on Artist {
      id
      name
      image {
        cropped(width: 100, height: 100) {
          url
        }
      }
    }
  `
)

export interface RelayProps {
  artist: {
    id: string | null
    name: string | null
    image: {
      cropped: {
        url: string | null
      } | null
    } | null
  }
}
