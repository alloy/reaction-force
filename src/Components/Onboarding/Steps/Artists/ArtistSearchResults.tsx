import * as React from "react"
import { commitMutation, createFragmentContainer, graphql, QueryRenderer, RelayProp } from "react-relay"

import { RecordSourceSelectorProxy, SelectorData } from "relay-runtime"
import { ContextConsumer, ContextProps } from "../../../Artsy"
import ItemLink from "../../ItemLink"

export interface Props {
  term: string
}

interface RelayProps extends React.HTMLProps<HTMLAnchorElement>, Props {
  relay?: RelayProp
  viewer: {
    match_artist: any[]
  }
}

class ArtistSearchResultsContent extends React.Component<RelayProps, null> {
  private excludedArtistIds: Set<string>

  constructor(props: RelayProps, context: any) {
    super(props, context)
    this.excludedArtistIds = new Set(this.props.viewer.match_artist.map(item => item._id))
  }

  onArtistFollowed(artistId: string, store: RecordSourceSelectorProxy, data: SelectorData): void {
    const suggestedArtist = store.get(data.followArtist.artist.related.suggested.edges[0].node.__id)
    this.excludedArtistIds.add(suggestedArtist.getValue("_id"))

    const popularArtistsRootField = store.get("client:root:viewer")
    const popularArtists = popularArtistsRootField.getLinkedRecords("match_artist", { term: this.props.term })
    const updatedPopularArtists = popularArtists.map(
      popularArtist => (popularArtist.getDataID() === artistId ? suggestedArtist : popularArtist)
    )

    popularArtistsRootField.setLinkedRecords(updatedPopularArtists, "match_artist", { term: this.props.term })
  }

  onFollowedArtist(artist: any) {
    commitMutation(this.props.relay.environment, {
      mutation: graphql`
        mutation ArtistSearchResultsArtistMutation($input: FollowArtistInput!, $excludedArtistIds: [String]!) {
          followArtist(input: $input) {
            artist {
              __id
              related {
                suggested(first: 1, exclude_followed_artists: true, exclude_artist_ids: $excludedArtistIds) {
                  edges {
                    node {
                      id
                      _id
                      __id
                      name
                      image {
                        cropped(width: 100, height: 100) {
                          url
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      `,
      variables: {
        input: {
          artist_id: artist.id,
          unfollow: false,
        },
        excludedArtistIds: Array.from(this.excludedArtistIds),
      },
      updater: (store: RecordSourceSelectorProxy, data: SelectorData) =>
        this.onArtistFollowed(artist.__id, store, data),
    })
  }

  render() {
    const artistItems = this.props.viewer.match_artist.map((artist, index) => (
      <ItemLink
        href="#"
        item={artist}
        key={index}
        id={artist.id}
        name={artist.name}
        image_url={artist.image && artist.image.cropped.url}
        onClick={() => this.onFollowedArtist(artist)}
      />
    ))

    return <div>{artistItems}</div>
  }
}

const ArtistSearchResultsContentContainer = createFragmentContainer(
  ArtistSearchResultsContent,
  graphql`
    fragment ArtistSearchResultsContent_viewer on Viewer {
      match_artist(term: $term) {
        id
        _id
        __id
        name
        image {
          cropped(width: 100, height: 100) {
            url
          }
        }
      }
    }
  `
)

const ArtistSearchResultsComponent: React.SFC<Props & ContextProps> = ({ term, relayEnvironment }) => {
  return (
    <QueryRenderer
      environment={relayEnvironment}
      query={graphql`
        query ArtistSearchResultsQuery($term: String!) {
          viewer {
            ...ArtistSearchResultsContent_viewer
          }
        }
      `}
      variables={{ term }}
      render={({ error, props }) => {
        if (props) {
          return <ArtistSearchResultsContentContainer viewer={props.viewer} term={term} />
        } else {
          return null
        }
      }}
    />
  )
}

export const ArtistSearchResults = ContextConsumer(ArtistSearchResultsComponent)
