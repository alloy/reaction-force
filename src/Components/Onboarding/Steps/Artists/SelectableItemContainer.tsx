import * as React from "react"
import * as Relay from "react-relay/classic"

import ItemLink from "./ItemLink"

interface Props {
  artists: any[]
}

class SelectableItemContainer extends React.Component<Props, null> {
  render() {
    const items = this.props.artists.map((artist, index) => <ItemLink href="#" artist={artist} key={index} />)

    return <div>{items}</div>
  }
}

export default Relay.createContainer(SelectableItemContainer, {
  fragments: {
    artists: () => Relay.QL`
      fragment on Artist @relay(plural: true) {
        ${ItemLink.getFragment("artist")}
      }
    `,
  },
})
