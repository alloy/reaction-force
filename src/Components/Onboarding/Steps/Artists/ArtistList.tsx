import * as React from "react"

import { ArtistSearchResults } from "./ArtistSearchResults"
// import PopularArtists from "./PopularArtists"

export interface Props {
  searchQuery: string
}

export default class ArtistList extends React.Component<Props, any> {
  render() {
    return <ArtistSearchResults term={this.props.searchQuery} />
    // return <PopularArtists />
  }
}
