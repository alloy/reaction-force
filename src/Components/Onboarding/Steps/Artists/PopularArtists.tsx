import * as React from "react"
import * as Relay from "react-relay"

import { artsyNetworkLayer } from "../../../../Relay/config"
import PopularArtistQueryConfig from "../../../../Relay/Queries/PopularArtist"

import SelectableItemContainer from "./SelectableItemContainer"

export interface RelayProps {
  popular_artists: {
    artists?: any[]
  }
}

export default class PopularArtists extends React.Component<null, null> {
  render() {
    return <PopularArtistContentList />
  }
}

function PopularArtistContentList() {
  Relay.injectNetworkLayer(artsyNetworkLayer())
  return <Relay.RootContainer Component={wrappedPopularArtistContent} route={new PopularArtistQueryConfig()} />
}

class PopularArtistsContent extends React.Component<RelayProps, null> {
  render() {
    return <SelectableItemContainer artists={this.props.popular_artists.artists} />
  }
}

const wrappedPopularArtistContent = Relay.createContainer(PopularArtistsContent, {
  fragments: {
    popular_artists: () => Relay.QL`
      fragment on PopularArtists {
        artists {
          ${SelectableItemContainer.getFragment("artists")}
        }
      }
    `,
  },
})
