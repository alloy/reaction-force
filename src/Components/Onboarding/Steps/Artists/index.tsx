import * as React from "react"
import styled from "styled-components"

import colors from "../../../../Assets/Colors"
import Icon from "../../../Icon"
import Input from "../../../Input"

import { StepProps } from "../../Types"
import { Layout } from "../Layout"
import ArtistList from "./ArtistList"

const OnboardingSearchBox = styled.div`
  width: 450px;
  margin: 0 auto 100px;
  border-bottom: 1px solid #e5e5e5;
`

interface State {
  inputText: string
}

export default class Artists extends React.Component<StepProps, State> {
  state = {
    inputText: "",
  }

  // onInputChange = e => {
  //   this.props.onStateChange({ nextButtonEnabled: true })
  // }

  searchTextChanged(e) {
    const updatedInputText = e.target.value
    this.setState({ inputText: updatedInputText })
  }

  render() {
    return (
      <Layout
        title="Follow a few artists that interest you most"
        subtitle="Follow one or more"
        onNextButtonPressed={null}
      >
        <OnboardingSearchBox>
          <Input
            placeholder={"Search artists..."}
            leftView={<Icon name="search" color={colors.graySemibold} />}
            block
            onInput={this.searchTextChanged.bind(this)}
            onPaste={this.searchTextChanged.bind(this)}
            onCut={this.searchTextChanged.bind(this)}
          />
          <div style={{ marginBottom: "35px" }} />
          <ArtistList searchQuery={this.state.inputText} />
        </OnboardingSearchBox>
      </Layout>
    )
  }
}
