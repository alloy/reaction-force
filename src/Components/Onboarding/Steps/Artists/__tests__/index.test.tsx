import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import Artists from "../"
import { ContextProvider } from "../../../../Artsy"

jest.mock("../../../../../Utils/metaphysics.ts", () => ({
  metaphysics: jest.fn(),
}))

describe("Artists", () => {
  describe("snapshot", () => {
    it("renders the artist step", () => {
      const component = renderer
        .create(
          <ContextProvider>
            <Artists onNextButtonPressed={jest.fn()} />
          </ContextProvider>
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
