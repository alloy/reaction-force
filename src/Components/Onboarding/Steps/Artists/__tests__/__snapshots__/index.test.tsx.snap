// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists snapshot renders the artist step 1`] = `
.c6 {
  font-family: artsy-icons;
  color: #666666;
  font-size: 24px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c7 {
  border: 0;
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  outline: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  padding: 10px;
  box-shadow: none;
  font-size: 17px;
  -webkit-transition: border-color .25s;
  transition: border-color .25s;
  margin-right: 10px;
  resize: none;
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  border: 2px solid #e5e5e5;
  -webkit-transition: border-color .25s;
  transition: border-color .25s;
  border: 0;
  padding: 12px;
  margin-right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c5:focus,
.c5.focused {
  border-color: #6e1fff;
  outline: 0;
}

.c5:disabled {
  border: 2px dotted #e5e5e5;
}

.c5 .border-container {
  z-index: -1;
  border: 2px solid #e5e5e5;
  -webkit-transition: border-color .25s;
  transition: border-color .25s;
  position: absolute;
  width: calc(100% - 4px);
  height: calc(100% - 4px);
  top: 0;
  left: 0;
}

.c5 .border-container:focus,
.c5 .border-container.focused {
  border-color: #6e1fff;
  outline: 0;
}

.c5 .border-container:disabled {
  border: 2px dotted #e5e5e5;
}

.c10 {
  background: #e5e5e5;
  color: rgba(0,0,0,0.5);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c10:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c9 {
  background: white;
  color: rgba(0,0,0,0.5);
  border: 1px solid #e5e5e5;
}

.c9:hover:not(:disabled) {
  background: white;
  color: #6e1fff;
}

.c2 {
  font-size: 50px;
  color: inherit;
  margin: 20px 0;
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
}

.c0 {
  max-width: 930px;
  margin-left: auto;
  margin-right: auto;
}

.c1 {
  text-align: center;
  line-height: normal;
}

.c3 {
  color: #999999;
  margin-bottom: 100px;
  text-align: center;
  line-height: normal;
}

.c8 {
  margin: 0 auto 50px;
  display: block;
  width: 250px;
}

.c4 {
  width: 450px;
  margin: 0 auto 100px;
  border-bottom: 1px solid #e5e5e5;
}

@media (max-width:48em) {
  .c2 {
    font-size: 25px;
  }
}

@media (max-width:48em) {
  .c0 {
    margin: 20px;
  }
}

@media (max-width:48em) {
  .c1 {
    text-align: left;
  }
}

@media (max-width:48em) {
  .c3 {
    text-align: left;
    margin-bottom: 15px;
    font-size: 20px;
  }
}

@media (max-width:48em) {
  .c8 {
    width: 100%;
  }
}

@media (max-width:48em) {
  .c4 {
    width: 100%;
    margin-bottom: 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    color="inherit"
  >
    Follow a few artists that interest you most
     
  </div>
  <div
    className="c3 c2"
    color="inherit"
  >
    Follow one or more
  </div>
  <div>
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        <div
          className="border-container"
        />
        <div
          className="c6"
          onClick={undefined}
          style={undefined}
        >
          î˜¬
        </div>
        <input
          className="c7"
          onBlur={[Function]}
          onCut={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onPaste={[Function]}
          placeholder="Search artists..."
          value={undefined}
        />
      </div>
      <div
        style={
          Object {
            "marginBottom": "35px",
          }
        }
      />
    </div>
  </div>
  <button
    className="c8 c9 c10"
    disabled={true}
    onClick={null}
  >
    <span>
      Next
    </span>
  </button>
</div>
`;
